### DISCUSSION TITLE
Is there any danger of being banned by using the WhatsApp client?

### DISCUSSION THREAD

**Discussion by melecoaze** [d99kris/nchat#147] [Q&A]
So I've been using nchat for telegram to send simple stuff to myself, but not many people I know use telegram.

I've been wanting to try the whatsapp version but I'm worried I might get banned, and I need whatsapp for work.

Is there any danger of being banned? Has anyone here been banned by using it?

**Comment by d99kris** [comment]
Hi @melecoaze - Since nchat added support for WhatsApp multi-device in April 2022 I have not heard of a single case of anyone getting banned using nchat, so I think it's fairly safe at the moment. nchat uses [whatsmeow](https://github.com/tulir/whatsmeow) as underlying library for WhatsApp support and in that project there are sometimes reports of [users](https://github.com/tulir/whatsmeow/issues/14) [getting](https://github.com/tulir/whatsmeow/issues/465) [banned](https://github.com/tulir/whatsmeow/issues/396). But I think in those cases users are effectively using whatsmeow to spam send messages (to non-contacts especially, and at high rate), so getting banned is not surprising. For normal usage I think it should be safe.

Having that said, I just recently read some article reporting that Meta was looking into monetizing WhatsApp further (no timeline announced though), so it would not be a surprise to me if they tighten up access for third-party apps at some point in the future (Meta is for example fairly effective in temporarily banning usage of third-party **Messenger** apps). Given the fact that there are several third-party WhatsApp apps I don't expect them to permanently ban such users, but rather just prevent access for the third-party WhatsApp apps or temporarily ban for some short time and then allow reactivating account using official app, in case they eventually tighten controls.

Anyway, it would be interesting to hear other users' experience so I hope others chip in, whether they encountered issues or not. 

If you do choose to use nchat, do try to update it every quarter or so, as the protocol (and whatsmeow) is updated continuosly, so WhatsApp may block access from old versions of the protocol.

**Edit:** Feel free to join the Telegram group https://t.me/nchatusers and ask other users there, as most users may not browse the Discussions section of this project on a regular basis.

**Reply by d99kris** [reply]
While I have not received any reports on `nchat` users encountering issues, today I saw an [whatsmeow issue](https://github.com/tulir/whatsmeow/issues/561) (the library nchat uses for whatsapp) where some users had been getting logged out from their official WhatsApp app with message:
```
You have been logged out for using an unofficial app.
Download the official WhatsApp and verify your account to log back in.
```
However googling the message I also found [other reports](https://discussions.apple.com/thread/255514715) which sound normal app users also encountered this (i.e. not necessarily using third-party clients), so maybe no need for concern.
